RecoTracker/TkTrackingRegions/plugins/AreaSeededTrackingRegionsBuilder.h

```diff
@@ -0,0 +1,108 @@
+#ifndef RecoTracker_TkTrackingRegions_AreaSeededTrackingRegionsBuilder_h
+#define RecoTracker_TkTrackingRegions_AreaSeededTrackingRegionsBuilder_h
+
+#include "RecoTracker/TkTrackingRegions/interface/RectangularEtaPhiTrackingRegion.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+class AreaSeededTrackingRegionsBuilder {
+public:
+  using Origin = std::pair<GlobalPoint, float>; // (origin, half-length in z)
+  using Origins = std::vector<Origin>;
+
+  class Area {
+  public:
+    // phimin and phimax, and hence xmin+xmax and ymin+ymax are
+    // ordered by which way one goes around the unit circle, so it may
+    // happen that actually phimax < phimin
+    Area(double rmin, double rmax, double phimin, double phimax, double zmin, double zmax):
+      m_zmin(zmin), m_zmax(zmax)
```

Comment:
is double precision required here?

Reply:
Very likely not, especially given that all the subsequent calculations in `AreaSeededTrackingRegionsBuilder` are done in float. I'll change these to `float`.